readthedocs/core/unresolver.py

```diff
@@ -0,0 +1,68 @@
+import logging
+from urllib.parse import urlparse
+from collections import namedtuple
+
+from django.urls import resolve as url_resolve
+from django.test.client import RequestFactory
+
+from readthedocs.core.utils.extend import SettingsOverrideObject
+from readthedocs.proxito.middleware import map_host_to_project_slug
+from readthedocs.proxito.views.utils import _get_project_data_from_request
```

Comment:
If we're using it outside of proxito, it probably shouldn't be private (underscore prefixed).